iotedgedev/modules.py

```diff
@@ -70,44 +83,39 @@ def build(self):
                         # i.e. when found: filter-module/Docker/linux-x64/Dockerfile.debug and CONTAINER_TAG = jong
                         # we'll get: filtermodule:linux-x64-debug-jong
 
-                        runtime = os.path.basename(
-                            os.path.dirname(docker_file))
-                        ext = "" if os.path.splitext(docker_file)[
-                            1] == "" else "-" + os.path.splitext(docker_file)[1][1:]
                         container_tag = "" if self.envvars.CONTAINER_TAG == "" else "-" + \
                             self.envvars.CONTAINER_TAG
 
-                        tag_name = runtime + ext + container_tag
+                        #tag_name = runtime + ext + container_tag
+                        tag_name = module_json.tag_version + container_tag
 
                         # construct the build output path
                         build_path = os.path.join(
-                            os.getcwd(), "build", "modules", module, runtime)
+                            os.getcwd(), "build", "modules", module)  # , runtime)
                         if not os.path.exists(build_path):
                             os.makedirs(build_path)
 
-                        # dotnet publish
+                        # publish module
                         self.output.info(
-                            "PUBLISHING PROJECT: " + project_files[0])
-
-                        self.utility.exe_proc(["dotnet", "publish", project_files[0], "-f", "netcoreapp2.0",
-                                               "-o", build_path, "-v", self.envvars.DOTNET_VERBOSITY])
+                            "PUBLISHING PROJECT: " + module_dir)
+                        mod_proc.publish(module_dir, build_path)
 
                         # copy Dockerfile to publish dir
                         build_dockerfile = os.path.join(
                             build_path, docker_file_name)
-
-                        copyfile(docker_file, build_dockerfile)
+                        copyfile(os.path.join(
+                            module_dir, docker_file_name), build_dockerfile)
 
                         image_destination_name = "{0}/{1}:{2}".format(
-                            self.envvars.CONTAINER_REGISTRY_SERVER, module, tag_name).lower()
+                            self.envvars.CONTAINER_REGISTRY_SERVER, module, module_json.tag_version, "-", arch_key, tag_name).lower()
```

Comment:
This doesn't seem right.  It should be modulename:version-containertag-arch.

Also with format...you only have 3 placeholders, but 6 parameters.